activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -79,16 +79,29 @@ def error_number(exception)
       # QUOTING ==================================================
       #++
 
+      # If this connection wasn't verified
+      # we might need to reconnect and try it again
       def quote_string(string)
-        @connection.escape(string)
+        with_connection_verify(Mysql2::Error) do
+          @connection.escape(string)
+        end
       end
 
       #--
       # CONNECTION MANAGEMENT ====================================
       #++
 
       def active?
-        @connection.ping
+        return false if !connected?
+
+        if @connection.ping
+          true
+        elsif @verify_connection
+          reconnect!
```

Comment:
I see the point here, but before my changes we called `connection.ping` in `active?` that would also reconnect if the connection is lost.